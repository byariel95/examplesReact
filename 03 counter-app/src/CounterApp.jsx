import React, {useState} from 'react';
import PropTypes from 'prop-types';

const CounterApp = ({ value=10 }) => {

    const [counter, setCounter] = useState(value);

    const handleAdd = () => setCounter(counter + 1);

    const substractAdd = () => setCounter(counter - 1);
    
    const resetvalue = () =>  setCounter( value);
    

    return (
        <>
            <h1>Counter </h1>
            <h2>{ counter }</h2> 
            <button onClick={handleAdd}>+ 1</button>
            <button onClick={substractAdd}>- 1</button>
            <button onClick={resetvalue}>reset</button>
        </>
    );


};

CounterApp.propTypes = {

    value: PropTypes.number,

}
 

export default  CounterApp;