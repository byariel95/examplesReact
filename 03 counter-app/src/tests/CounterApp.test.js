
import React from 'react';
import '@testing-library/jest-dom';
import {shallow } from 'enzyme';
import CounterApp from '../CounterApp';

describe('pruebas en el componente counter', () => {
    
    let wrapper= shallow( <CounterApp  />);

    // se ejecuta entes de  cada evaluacion
    beforeEach(()=>{

        wrapper = shallow( <CounterApp  />);
    })
    
    test('debera mostrar el snapshot del componente counter', () => {
        
       
        expect( wrapper ).toMatchSnapshot();
    });
    
    test('debera mostrar el valor de 100', () => {
        const valorDefecto = 100;
        const wrapper = shallow( 
        <CounterApp 
        value={valorDefecto} 
        />
        );

        const counter = wrapper.find('h2').text().trim();
        expect(counter).toBe(`${valorDefecto}`);
    });
    
    test('debera simular los clicks y aumentar', () => {
     
        wrapper.find('button').at(0).simulate('click'); // boton numero 1
        const counterText = wrapper.find('h2').text().trim();  
        expect(counterText).toBe('11');
       
        
    });
    
    test('debera simular los clicks y disminuir', () => {
     
        wrapper.find('button').at(1).simulate('click'); // boton numero 1
        const counterText = wrapper.find('h2').text().trim();  
        expect(counterText).toBe('9');
       
        
    });


    test('debera de establecer el valor por defecto', () => {
     
        const valorDefecto = 100;
        const wrapper = shallow( <CounterApp value={valorDefecto} />);
        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(2).simulate('click'); 
        const counterText = wrapper.find('h2').text().trim();  
        expect(counterText).toBe(valorDefecto.toString());
       
        
    });
    

})
