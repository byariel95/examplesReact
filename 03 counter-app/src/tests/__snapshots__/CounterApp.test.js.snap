// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en el componente counter debera mostrar el snapshot del componente counter 1`] = `
<Fragment>
  <h1>
    Counter 
  </h1>
  <h2>
    10
  </h2>
  <button
    onClick={[Function]}
  >
    + 1
  </button>
  <button
    onClick={[Function]}
  >
    - 1
  </button>
  <button
    onClick={[Function]}
  >
    reset
  </button>
</Fragment>
`;
